promise.go

```diff
@@ -51,6 +53,16 @@ func (np *NodePromise) Send(nd Node) {
 	close(np.done)
 }
 
+// Poll returns the result of the promise if ready but doesn't block.
+func (np *NodePromise) Poll() (Node, error) {
```

Comment:
I'm not sure about the name.
poll (2) in Linux is blocking.

Reply:
Both java and rust use a poll function to check futures without waiting.

Reply:
Hmm, interesting.